src/harness/unittests/compileOnSave.ts

```diff
@@ -285,225 +339,218 @@ namespace ts.projectSystem {
-                const typingsInstaller = createTestTypingsInstaller(host);
-                const session = createSession(host, typingsInstaller);
                 openFilesForSession([moduleFile1], session);
-                sendAffectedFileRequestAndCheckResult(session, moduleFile1FileListRequest, []);
+
+                sendAffectedFileRequestAndCheckResult(session,
+                    { projectFileName: configFile.path, file: moduleFile1.path },
+                    []);
             });
 
             it("should return empty array if noEmit is set", () => {
-                configFile = {
-                    path: "/a/b/tsconfig.json",
-                    content: `{
-                        "compileOnSave": true,
-                        "compilerOptions": {
-                            "noEmit": true
-                        }
-                    }`
-                };
-
-                const host = createServerHost([moduleFile1, file1Consumer1, file1Consumer2, configFile, libFile]);
+                const host = new fakes.FakeServerHost({ lib: true });
+                const configFile = host.vfs.addFile("/a/b/tsconfig.json", `{ "compileOnSave": true, "compilerOptions": { "noEmit": true } }`);
+                const moduleFile1 = host.vfs.addFile("/a/b/moduleFile1.ts", `export function Foo() { };`);
+                host.vfs.addFile("/a/b/file1Consumer1.ts", `import {Foo} from "./moduleFile1"; export var y = 10;`);
+                host.vfs.addFile("/a/b/file1Consumer2.ts", `import {Foo} from "./moduleFile1"; let z = 10;`);
+
                 const typingsInstaller = createTestTypingsInstaller(host);
                 const session = createSession(host, typingsInstaller);
                 openFilesForSession([moduleFile1], session);
-                sendAffectedFileRequestAndCheckResult(session, moduleFile1FileListRequest, []);
+                sendAffectedFileRequestAndCheckResult(session,
+                    { projectFileName: configFile.path, file: moduleFile1.path },
+                    []);
             });
 
             it("should save when compileOnSave is enabled in base tsconfig.json", () => {
-                configFile = {
-                    path: "/a/b/tsconfig.json",
-                    content: `{
-                        "extends": "/a/tsconfig.json"
-                    }`
-                };
-
-                const configFile2: FileOrFolder = {
-                    path: "/a/tsconfig.json",
-                    content: `{
-                        "compileOnSave": true
-                    }`
-                };
-
-                const host = createServerHost([moduleFile1, file1Consumer1, file1Consumer2, configFile2, configFile, libFile]);
-                const typingsInstaller = createTestTypingsInstaller(host);
-                const session = createSession(host, typingsInstaller);
+                const host = new fakes.FakeServerHost({ lib: true });

```

Comment:
You could probably reduce the diff in changing these tests just by adding another parameter to the `FakeServerHost` constructor which takes a list of fake file objects to add to the FS. Especially since so many of these tests start with "make host, add a bunch of files".